src/stan/services/util/initialize.hpp

```diff
@@ -13,202 +13,211 @@
-            continue;
-          }
-          std::stringstream log_prob_msg;
-          std::vector<double> gradient;
-          bool gradient_ok = true;
-          clock_t start_check = clock();
-          try {
-            log_prob = stan::model::log_prob_grad<true, true>
-              (model, unconstrained, disc_vector,
-               gradient, &log_prob_msg);
-          } catch (const std::exception& e) {
-            if (log_prob_msg.str().length() > 0)
-              logger.info(log_prob_msg);
-            logger.info(e.what());
-            throw;
-          }
-          clock_t end_check = clock();
-          double deltaT = static_cast<double>(end_check - start_check)
-            / CLOCKS_PER_SEC;
-          if (log_prob_msg.str().length() > 0)
-            logger.info(log_prob_msg);
+      if (!any_initialized) {
+        unconstrained = random_context.get_unconstrained();
+      } else {
+        stan::io::chained_var_context context(init, random_context);
 
-          for (size_t i = 0; i < gradient.size(); ++i) {
-            if (gradient_ok && !boost::math::isfinite(gradient[i])) {
-              logger.info("Rejecting initial value:");
-              logger.info("  Gradient evaluated at the initial value"
-                          " is not finite.");
-              logger.info("  Stan can't start sampling from this"
-                          " initial value.");
-              gradient_ok = false;
-            }
-          }
-          if (gradient_ok && print_timing) {
-            logger.info("");
-            std::stringstream msg1;
-            msg1 << "Gradient evaluation took " << deltaT << " seconds";
-            logger.info(msg1);
+        model.transform_inits(context,
+                              disc_vector,
+                              unconstrained,
+                              &msg);
+      }
+    } catch (std::domain_error& e) {
+      if (msg.str().length() > 0)
+        logger.info(msg);
+      logger.info("Rejecting initial value:");
+      logger.info("  Error evaluating the log probability"
+                  " at the initial value.");
+      logger.info(e.what());
+      continue;
+    } catch (std::exception& e) {
+      if (msg.str().length() > 0)
+        logger.info(msg);
+      logger.info("Unrecoverable error evaluating the log probability"
+                  " at the initial value.");
+      logger.info(e.what());
+      throw;
+    }
 
-            std::stringstream msg2;
-            msg2 << "1000 transitions using 10 leapfrog steps"
-                 << " per transition would take"
-                 << " " << 1e4 * deltaT << " seconds.";
-            logger.info(msg2);
+    msg.str("");
+    double log_prob(0);
+    try {
+      log_prob = model.template log_prob<false, true>
+                 (unconstrained, disc_vector, &msg);
+      if (msg.str().length() > 0)
+        logger.info(msg);
+    } catch (std::domain_error& e) {
+      if (msg.str().length() > 0)
```

Comment:
[suggestion]

This pattern is everywhere, and while it's simple, I'd rather not see this test everywhere.  Can we just have the `info()` mthod ignore empty messages in general?  If we really want a blank line, we can pass `info(" ")` or `info("\n")` if it doesn't build one in automatically.

If not that, then maybe wrap this all up in a function that does the test so this pattern isn't cut-and-paste dozens of times.

Reply:
The pattern is everywhere. I'll create an issue and we'll take a pass.

Reply:
Created an issue.